stream-access-control/access-control-core/src/main/java/com/backbase/stream/service/UserService.java

```diff
@@ -424,6 +425,39 @@ private Optional<UserRequestBody> updateRequired(EnhancedUserRepresentation curr
         return updateRequired ? Optional.of(userRequestBody) : Optional.empty();
     }
 
+
+    /**
+     * Update Identity User, ex: emailAddress, mobileNumber, attributes, and additions
+     *
+     * @param user
+     * @return Mono<User>
+     */
+    public Mono<User> updateIdentity(User user) {
+
+        Objects.requireNonNull(user.getInternalId(), "user internalId is required");
+
+        return identityManagementApi.getIdentity(user.getInternalId())
+                .map(mapper::mapUpdateIdentity)
+                .flatMap(updateIdentityRequest -> {
+                    log.debug("Trying to update identity attributes and additions, externalId [{}]", user.getExternalId());
+                        if (updateIdentityRequest.getAttributes() == null) {
+                            updateIdentityRequest.attributes(new HashMap<>());
+                        }
+                        if (updateIdentityRequest.getAdditions() == null) {
+                            updateIdentityRequest.additions(new HashMap<>());
+                        }
+                        updateIdentityRequest.getAttributes().putAll(requireNonNullElse(user.getAttributes(), Map.of()));
+                        updateIdentityRequest.getAdditions().putAll(requireNonNullElse(user.getAdditions(), Map.of()));
```

Comment:
Do you see any concern, if you replace lines 443-450 with

```
updateIdentityRequest.attributes(user.getAttributes());
updateIdentityRequest.additions(user.getAdditions());
```

Reply:
yeah, there is a concern that it might replace all original user attributes. so the current logic mantain the original one and just add or replace the new attributes.

Reply:
yeah, I see your point.